modin/core/dataframe/pandas/partitioning/partition.py

```diff
@@ -273,13 +274,42 @@ def length(self):
         int
             The length of the object.
         """
+        self.try_build_length_cache()
+        return self._length_cache
+
+    def try_build_length_cache(self) -> Union[Any, int]:
+        """
+        Attempt to set this partition's length cache, and return it.
+
+        Returns
+        -------
+        Any | int
+            Either a future-like object representing the length of the object wrapped by this
+            partition, or the concrete value of the length if it was already cached or was
+            just computed.
+        """
         if self._length_cache is None:
             cls = type(self)
             func = cls._length_extraction_fn()
             preprocessed_func = cls.preprocess_func(func)
             self._length_cache = self.apply(preprocessed_func)
         return self._length_cache
 
+    def try_set_length_cache(self, length: int):
```

Comment:
nit: this function doesn't really "try" anything. Could it be named just `set_length_cache`? The same applies for `try_set_width_cache`.

Reply:
changed